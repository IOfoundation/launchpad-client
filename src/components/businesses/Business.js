import React, {PureComponent} from 'react';
import Chip from '../shared/Chip';
import FacebookIcon from '../shared/FacebookIcon';
import TwitterIcon from '../shared/TwitterIcon';
import LinkedinIcon from '../shared/LinkedinIcon';

import {PropTypes} from 'prop-types';
import {isEmpty} from 'lodash';

class Business extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      expanded: this.props.expanded,
    };
  }

  toggleCard() {
    this.setState({expanded: !this.state.expanded});
  }

  _renderContacts(subject) {
    return (
      <div className="col-lg-4 col-md-4 col-xs-4 p-0 main-contact">
        <p className="business-title">{'Contact:'}</p>
        {isEmpty(subject.name) ? '' : <h4>{subject.name}</h4>}
        {isEmpty(subject.phones) ? '' : <h4>{subject.phones[0].number}</h4>}
        {isEmpty(subject.email) ? '' : <h4>{subject.email}</h4>}
      </div>
    );
  }
  _renderOtherLocations(otherLocations) {
    return (
      <div className="col-lg-12 col-md-12 col-xs-12 p-0 grid">
        {otherLocations.map(location => {
          return (
            <div key={location.id} className="col-lg-12 col-md-12 col-xs-12 m-bot-24 p-0 grid">
              <div className="col-lg-6 col-md-6 col-xs-6 p-0 m-right-54">
                <h4>{location.address.address_1}</h4>
                {location.address.address_2 && <h4>{location.address.address_2}</h4>}
                <h4>
                  {`${location.address.city}, ${location.address.state_province} ${location.address.postal_code}`}
                </h4>
              </div>
              <div className="col-lg-4 col-md-4 col-xs-6 p-0 m-top-24">
                {location.phones.length > 0 && <h4>{location.phones[0].number}</h4>}
                {location.email && <h4>{location.email}</h4>}
              </div>
            </div>
          );
        })}
      </div>
    )
  }
  render() {
    const {business, isSelected} = this.props;
    const locations = business.locations;
    const [main_location, ...other_locations] = locations;
    return (
      <div className="business-card" style={{backgroundColor: isSelected ? '#E5E5E5' : '#F2F2F2'}}>
        <div
          className={
            this.state.expanded
              ? 'business-card-expand grid'
              : 'business-card-collapse grid'
          }
        >
          <div
            className={
              this.state.expanded
                ? 'business col-lg-2 col-md-2 col-xs-2 p-0 m-bot-20'
                : 'business business-img'
            }
          >
            <img
              className="business-logo"
              src="../static-data/images/orgs-placeholder.png"
            />
          </div>
          <div
            className={
              this.state.expanded
                ? 'business col-lg-10 col-md-10 col-xs-10 p-right-0'
                : 'business col-lg-12 col-md-12 col-xs-12 p-0'
            }
          >
            <div className="preview-details-container">
              <h3 className="title m-bot-8">
                {business.name}
              </h3>
              <p className="preview-details">{business.description}</p>
              <div className={this.state.expanded ? ('col-lg-12 social-icons p-0 m-top-20') : ('social-icons-hide')}>
                {business.facebook ? <a href={business.facebook}><FacebookIcon className={'icon-svg'} size={18} /></a> : ''}
                {business.twitter ? <a href={business.twitter}><TwitterIcon className={'icon-svg'} size={18} /></a> : ''}
                {business.linkedin ? <a href={business.linkedin}><LinkedinIcon className={'icon-svg'} size={18} /></a> : ''}
              </div>
            </div>
            <img
              className="business-card-icon"
              onClick={e => this.toggleCard(e)}
              style={{float: 'right'}}
              src={
                this.state.expanded
                  ? 'static-data/images/collapse-icon.png'
                  : 'static-data/images/expand-icon.png'
              }
            />
          </div>
          <div className="grid col-lg-12 col-md-12 col-xs-12 full-information p-0">
            <div className="grid col-lg-12 col-md-12 col-xs-12 p-0 m-bot-25">
              <div className="col-lg-4 col-md-4 col-xs-4 p-0 m-right-52 main-location">
                <p className="business-title">{'Main Location:'}</p>
                <h4>{main_location.address.address_1}</h4>
                {main_location.address.address_2 ? <h4>{main_location.address.address_2}</h4> : ''}
                <h4>
                  {`${main_location.address.city}, ${main_location.address.state_province}`}
                </h4>
              </div>
              {!isEmpty(business.contacts) && this._renderContacts(business.contacts[0])}
            </div>
            <hr />
            <p className="business-title col-lg-12 col-md-12 col-xs-12 p-0">
              {'Services:'}
            </p>
            {business.services.map(service => {
              return (
                <div
                  key={service.id}
                  className="col-lg-12 col-md-12 col-xs-12 grid p-0 business-service m-bot-28"
                >
                  <div className="col-lg-6 col-md-6 col-xs-6 p-left-0 p-right-4">
                    <h4 className='text-bold'>{service.name}</h4>
                    <p>{service.description}</p>
                  </div>
                  {!isEmpty(service.contacts) && this._renderContacts(service.contacts[0])}
                  {service.categories ?
                    <div className="col-lg-12 col-md-12 col-xs-12 p-0 m-top-16">
                      {service.categories.map(category => {
                        return (
                          <Chip
                            key={category.id}
                            text={category.name}
                            canDelete={false}
                          />
                        );
                      })}
                    </div>
                    : '' }
                </div>
              );
            })}
            <hr />
            {isEmpty(other_locations) ?
              '' :
              <div className="col-lg-12 col-md-12 col-xs-12 grid p-0">
                <div className="col-lg-12 col-md-12 col-xs-12 grid p-0">
                  <div className="col-lg-6 col-md-6 col-xs-6 p-0 m-right-54">
                    <p className="business-title">{'Other Locations:'}</p>
                  </div>
                  <div className="col-lg-4 col-md-4 col-xs-6 p-0">
                    <p className="business-title">{'Contact:'}</p>
                  </div>
                  {this._renderOtherLocations(other_locations)}
                </div>
              </div>
            }
          </div>
          <p className="location text-bold">
            <span>
              {locations.length}
              {locations.length == 1 ? ' location' : ' locations'}</span>
            <span className="m-x-7">{'|'}</span>
            <span>
              {locations.length == 1 ? '' : 'Main location in '}
              {main_location.address.city}
              {', '}
              {main_location.address.state_province}
            </span>
          </p>
        </div>
      </div>
    );
  }
}

Business.propTypes = {
  business: PropTypes.object.isRequired,
  expanded: PropTypes.bool,
  selectedColor: PropTypes.string.isRequired,
};

export default Business;
