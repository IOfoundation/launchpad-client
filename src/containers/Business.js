import React, {Component} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {PropTypes} from 'prop-types';

import MainLayout from '../components/layouts/Main';
import Logo from '../components/shared/Logo';
import FilterBox from '../components/filters/FilterBox';
import BusinessView from '../components/business/Main';
import * as actions from '../actions/business';

export class Business extends Component {
  componentWillMount(_nextProps) {
    const organizationId = this.props.params.id;
    this.props.actions.fetchOrganization(organizationId);
    this.props.actions.fetchFilterOptions();
  }

  handleTextSearchBusinesses(businessName) {
    const businessesFilters = this.props.location.query;
    this.props.actions.filterBusinessesByName(businessName, businessesFilters);
  }

  handleOnChangeFilterOptions(event) {
    event.preventDefault();
    const businessesFilters = this.props.location.query;
    const filterValue = event.target.id;

    this.props.actions.filterBusinesses(filterValue, businessesFilters);
  }

  render() {
    return (
      <MainLayout>
        <section>
          <div className="search-nav">
            <div className="row contentContainer between-xs">
              <div className="col-xs-3 search-nav_logo noPadding">
                <Logo />
              </div>
              <div className="col-xs-9 search-nav_filters noPadding">
                <FilterBox
                  handleTextSearchBusinesses={this.handleTextSearchBusinesses.bind(
                    this
                  )}
                  filterOptions={this.props.filters}
                  handleOnChangeFilterOptions={this.handleOnChangeFilterOptions.bind(
                    this
                  )}
                />
              </div>
            </div>
          </div>
          <BusinessView business={this.props.business} />
        </section>
      </MainLayout>
    );
  }
}

Business.propTypes = {
  actions: PropTypes.object,
  business: PropTypes.object.isRequired,
  filters: PropTypes.object,
  location: PropTypes.object,
  params: PropTypes.object,
};

const mapStateToProps = _state => {
  const {businesses} = _state;
  return {
    businesses: businesses.businesses,
    filters: businesses.filters,
    metadata: businesses.metadata,
    business: businesses.business,
  };
};

const mapDispatchToProps = _dispatch => {
  return {
    actions: bindActionCreators(actions, _dispatch),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Business);
