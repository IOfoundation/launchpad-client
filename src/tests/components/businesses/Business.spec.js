import React from 'react';
import {shallow} from 'enzyme';

import Business from '../../../components/businesses/Business';

const handleClickOnBusiness = jest.fn();

function mockOrganization() {
  return {
    id: 1,
    name: 'organization1',
    description: 'orgDescription',
    email: 'jondoe@something.org',
    contacts: [],
    phones: [
      {
        id: 1,
        number: '(123) 456 - 7890',
      }
    ],
    locations: [
      {
        id: 1,
        name: 'location1',
        address: {
          address_1: '123 Road St',
          address_2: null,
          city: 'Sacramento',
          state_province: 'CA',
          postal_code: '12345'
        }
      },
      {
        id: 2,
        name: 'location2',
        address: {
          address_1: '456 Street Rd',
          address_2: null,
          city: 'Sacramento',
          state_province: 'CA',
          postal_code: '12345'
        }
      },
    ],
    services: [
      {
        id: 1,
        name: 'service1',
        description: 'lorem ipsum...',
        email: null,
        phones: [],
        categories: [
          {
            id: 54,
            name: 'category1',
          },
          {
            id: 55,
            name: 'category2',
          },
          {
            id: 56,
            name: 'category3',
          },
        ],
      },
      {
        id: 2,
        name: 'service2',
        description: 'lorem ipsum...',
        email: null,
        phones: [],
        categories: [
          {
            id: 54,
            name: 'category1',
          },
        ],
      }
    ],
  };
}

describe('<Business />', () => {
  it('Renders the name of the Business', () => {
    const organization = mockOrganization();
    const wrapper = shallow(
      <Business
        business={organization}
        key={organization.id}
        handleClickOnBusiness={handleClickOnBusiness}
      />
    );
    expect(wrapper.find('.title').text()).toBe('organization1');
  });

  it('Renders the description of the Organization', () => {
    const organization = mockOrganization();
    const wrapper = shallow(
      <Business
        business={organization}
        key={organization.id}
        handleClickOnBusiness={handleClickOnBusiness}
      />
    );
    expect(wrapper.find('.preview-details').text()).toBe('orgDescription');
  });

  it('Renders a Main Location for an Organization', () => {
    const organization = mockOrganization();
    const wrapper = shallow(
      <Business
        business={organization}
        key={organization.id}
        handleClickOnBusiness={handleClickOnBusiness}
      />
    );
    expect(wrapper.find('.main-location').contains('123 Road St')).toBe(true);
  });

  it('Renders a Main Contact for the Organization', () => {
    const organization = mockOrganization();
    const wrapper = shallow(
      <Business
        business={organization}
        key={organization.id}
        handleClickOnBusiness={handleClickOnBusiness}
      />
    );
    expect(wrapper.find('.main-contact').contains('jondoe@something.org')).toBe(true);
  });

  it('Renders a list of services offered', () => {
    const organization = mockOrganization();
    const wrapper = shallow(
      <Business
        business={organization}
        key={organization.id}
        handleClickOnBusiness={handleClickOnBusiness}
      />
    );
    expect(wrapper.find('.business-service').length).toEqual(2);
  });

  it('Update expand/collapse state when user clicks expand button', () => {
    const organization = mockOrganization();
    const wrapper = shallow(
      <Business
        business={organization}
        key={organization.id}
        handleClickOnBusiness={handleClickOnBusiness}
      />
    );
    wrapper.find('.business-card-icon').simulate('click');
    expect(wrapper.state('expanded')).toEqual(true);
  });
});
