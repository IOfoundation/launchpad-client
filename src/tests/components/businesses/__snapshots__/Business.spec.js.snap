// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Business /> renders snapshot of Business 1`] = `
<div
  className="business-card"
  style={
    Object {
      "backgroundColor": "#F2F2F2",
    }
  }
>
  <div
    className="business-card-expand grid"
  >
    <div
      className="business col-lg-2 col-md-3 col-xs-12 p-0 m-bot-20"
      onClick={[Function]}
    >
      <img
        className="business-logo"
        src="../static-data/images/orgs-placeholder.png"
      />
    </div>
    <div
      className="business expanded col-lg-10 col-md-9 col-xs-12 p-right-0"
    >
      <div
        className="business-details-container"
      >
        <div
          className="business-information"
          onClick={[Function]}
        >
          <h3
            className="title m-bot-8"
          >
            organization1
          </h3>
          <p
            className="business-description"
          >
            orgDescription
          </p>
        </div>
        <div
          className="col-lg-12 social-icons p-0 m-top-20"
        />
      </div>
      <img
        className="business-card-icon"
        onClick={[Function]}
        src="static-data/images/collapse-icon.png"
        style={
          Object {
            "float": "right",
          }
        }
      />
    </div>
    <div
      className="grid col-lg-12 col-md-12 col-xs-12 full-information p-0"
    >
      <div
        className="grid col-lg-12 col-md-12 col-xs-12 p-0 m-bot-25"
      >
        <div
          className="col-lg-4 col-md-4 col-xs-12 p-0 m-right-52 main-location"
        >
          <p
            className="business-title"
          >
            Main Location:
          </p>
          <h4>
            123 Road St
          </h4>
          <h4>
            Sacramento, CA
          </h4>
        </div>
        <div
          className="col-lg-5 col-md-5 col-xs-12 p-0 main-contact"
        >
          <p
            className="business-title"
          >
            Contact:
          </p>
          <h4>
            (530) 795-1520
          </h4>
          <h4
            className="break-word"
          >
            jondoe@test.org
          </h4>
        </div>
      </div>
      <hr />
      <p
        className="business-title col-lg-12 col-md-12 col-xs-12 p-0"
      >
        Services
      </p>
      <div
        className="col-lg-12 col-md-12 col-xs-12 grid p-0 business-service m-bot-28"
        key="1"
      >
        <div
          className="service-description col-lg-12 col-md-12 col-xs-12 p-left-0"
        >
          <h4
            className="text-bold"
          >
            service1
          </h4>
          <p>
            lorem ipsum...
          </p>
        </div>
        <div
          className="col-lg-12 col-md-12 col-xs-12 p-0 m-top-16"
        >
          <Chip
            canDelete={false}
            key="54"
            text="category1"
          />
          <Chip
            canDelete={false}
            key="55"
            text="category2"
          />
          <Chip
            canDelete={false}
            key="56"
            text="category3"
          />
        </div>
      </div>
      <div
        className="col-lg-12 col-md-12 col-xs-12 grid p-0 business-service m-bot-28"
        key="2"
      >
        <div
          className="service-description col-lg-12 col-md-12 col-xs-12 p-left-0"
        >
          <h4
            className="text-bold"
          >
            service2
          </h4>
          <p>
            lorem ipsum...
          </p>
        </div>
        <div
          className="col-lg-12 col-md-12 col-xs-12 p-0 m-top-16"
        >
          <Chip
            canDelete={false}
            key="54"
            text="category1"
          />
        </div>
      </div>
      <div
        className="col-lg-12 col-md-12 col-xs-12 grid p-0"
      >
        <div
          className="col-lg-6 col-md-6 col-xs-6 p-0 m-right-54"
        >
          <p
            className="business-title"
          >
            Other Locations
          </p>
        </div>
        <div
          className="col-lg-4 col-md-4 col-xs-6 p-0"
        >
          <p
            className="business-title"
          >
            Contact:
          </p>
        </div>
        <div
          className="col-lg-12 col-md-12 col-xs-12 p-0 grid"
        >
          <div
            className="col-lg-12 col-md-12 col-xs-12 m-bot-24 p-0 grid"
            key="2"
          >
            <div
              className="col-lg-6 col-md-6 col-xs-6 p-0 m-right-54"
            >
              <h4>
                456 Street Rd
              </h4>
              <h4>
                Sacramento, CA 12345
              </h4>
            </div>
            <div
              className="col-lg-4 col-md-4 col-xs-6 p-0 m-top-24"
            >
              <h4>
                (123) 456 - 7890
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
    <p
      className="location text-bold"
    >
      <span>
        2  Locations
      </span>
      <span
        className="m-x-7"
      >
        |
      </span>
      <span>
        Main location in 
        Sacramento
        , 
        CA
      </span>
    </p>
  </div>
</div>
`;
